kin-devplatform-sdk/src/main/java/kin/devplatform/KinEcosystemInitiator.java

```diff
@@ -0,0 +1,265 @@
+package kin.devplatform;
+
+import android.content.Context;
+import android.os.Handler;
+import android.os.Looper;
+import android.support.annotation.NonNull;
+import android.text.TextUtils;
+import java.util.UUID;
+import kin.core.KinClient;
+import kin.core.ServiceProvider;
+import kin.devplatform.accountmanager.AccountManager;
+import kin.devplatform.accountmanager.AccountManager.AccountState;
+import kin.devplatform.accountmanager.AccountManagerImpl;
+import kin.devplatform.accountmanager.AccountManagerLocal;
+import kin.devplatform.base.ObservableData;
+import kin.devplatform.base.Observer;
+import kin.devplatform.bi.EventLogger;
+import kin.devplatform.bi.EventLoggerImpl;
+import kin.devplatform.bi.events.GeneralEcosystemSdkError;
+import kin.devplatform.bi.events.KinSdkInitiated;
+import kin.devplatform.core.util.DeviceUtils;
+import kin.devplatform.core.util.ExecutorsUtil;
+import kin.devplatform.data.auth.AuthLocalData;
+import kin.devplatform.data.auth.AuthRemoteData;
+import kin.devplatform.data.auth.AuthRepository;
+import kin.devplatform.data.blockchain.BlockchainSourceImpl;
+import kin.devplatform.data.blockchain.BlockchainSourceLocal;
+import kin.devplatform.data.offer.OfferRemoteData;
+import kin.devplatform.data.offer.OfferRepository;
+import kin.devplatform.data.order.OrderLocalData;
+import kin.devplatform.data.order.OrderRemoteData;
+import kin.devplatform.data.order.OrderRepository;
+import kin.devplatform.exception.BlockchainException;
+import kin.devplatform.exception.KinEcosystemException;
+import kin.devplatform.network.model.AuthToken;
+import kin.devplatform.network.model.SignInData;
+import kin.devplatform.util.ErrorUtil;
+
+public final class KinEcosystemInitiator {
```

Comment:
As i understand this class can be run on a background thread and i saw that there is some state in this class.
My question is: 
Is this class thread safe? 
If not then did you check all kinds of scenarios that maybe can cause some issues?

Reply:
well, actually it's not, I didn't pay attention to this. internally we call it from main thread only (activity onCreate) externally, who knows?...
but, I'm not sure what's the correct approach here, most of the init is synchronized, and login again does no harm, so worst case login will be happen twice.
what should we do in case of 2 subsequent calls? block the second and notify when the first finishes? we can't just put `synchronize` here as it won't have the correct effect (there are callbacks here that don't adhere to the synchronized block, so still 2 calls can happen concurrently).
The best and simple practice is to let the developer to be aware to call it from the main thread and ensure it happens only once in his code, o.w. it will be hard from our side to prevent weird edge cases, and sync multiple calls.

Reply:
I will this on docs.

Reply:
Ok, thanks for the explanation.